dist/stat_data.py

```diff
@@ -277,7 +279,11 @@ def __init__(self, name, desc, flags=''):
         'no_aggregate,no_clear,no_scale'),
     TxnStat('txn_commit', 'transactions committed'),
     TxnStat('txn_fail_cache', 'transaction failures due to cache overflow'),
+    TxnStat('txn_not_visible_checkpoint',
```

Comment:
I'm not convinced about the utility of this pair of statistics. It was interesting for me while testing, and I think it indicates cases where visibility checks would have failed without this optimization. Adding a stat incr call into every visibility check has a cost.